### DISCUSSION TITLE
InputNumber Step rounds numbers

### DISCUSSION THREAD

**Discussion by michaelcsikos** [ant-design-blazor/ant-design-blazor#2678] [Q&A]
Setting a `Step` on the `InputNumber` rounds the value to the nearest step when tabbing out of the control. I expected the Step to only be a convenience to make it easier to jump by common increments, but the user can still enter a specific number with higher precision.

**Comment by ElderJames** [comment]
What do you mean, please?

**Comment by michaelcsikos** [comment]
Using a `Step` value causes the value to be rounded. A step should only be a convenience for the user with the up/down controls; it should not truncate a value if it has been entered. See the following example:
![InputNumber step](https://user-images.githubusercontent.com/21806535/211694332-7b6e2c93-6a46-4a1e-acc3-d16e981c0859.gif)
```razor
No step:
<br />
<AntDesign.InputNumber @bind-Value = "_valueNoStep" />
<br />
<br />
Step = 2.0
<br />
<AntDesign.InputNumber @bind-Value = "_valueWithStep" Step = "2.0" />

@code
{
    double _valueNoStep   { get; set; }
    double _valueWithStep { get; set; }
}
```
